app/models/space.rb

```diff
@@ -16,7 +16,12 @@ class Space < ActiveRecord::Base
 
   scope :is_private, -> { where(is_private: true) }
   scope :is_public, -> { where(is_private: false) }
-  scope :visible_by, -> (user) { where 'is_private IS false OR user_id = ?', user.try(:id) }
+  scope :public_or_belonging_to, -> (user) { where 'is_private IS false OR user_id = ?', user.try(:id) }
+
+  def self.visible_by(user)
+    return public_or_belonging_to(user) if user.organization.nil?
```

Comment:
Small comment, but this first line is a bit un-ruby like, if this is a typical case (that there's no organization).  These short statements are typically for catches / errors and stuff, while this seems like more important logic.